internal/engine/upper.go

```diff
@@ -663,8 +664,8 @@ func handlePodChangeAction(ctx context.Context, state *store.EngineState, pod *v
 	checkForPodCrash(ctx, ms, *podInfo)
 
 	if int(cStatus.RestartCount) > podInfo.ContainerRestarts {
-		podInfo.PreRestartLog = append([]byte{}, podInfo.CurrentLog...)
-		podInfo.CurrentLog = []byte{}
+		podInfo.PreRestartLog = podInfo.CurrentLog.Copy()
```

Comment:
i'm not sure that you need a Copy() method? presumably, it it's append only, normal struct copying will do what we want (retain a pointer to the byte-array and the length of the slice)